spec/controllers/api/v3/patients_controller_spec.rb

```diff
@@ -287,6 +287,29 @@ def create_record_list(n, options = {})
                  deleted_reason: 'duplicate')
       end
 
+        it 'calls #discard_data on a a patient when the patient payload has the deleted_at field set' do
+          patient = instance_double("Patient")
+          phone_number = create(:patient_phone_number, patient: existing_patient)
+          pbi = create(:patient_business_identifier, patient: existing_patient)
+          allow(Patient).to receive_messages(find: patient, merge: patient)
+          allow(PatientPhoneNumber).to receive(:merge).and_return(phone_number)
+          allow(PatientBusinessIdentifier).to receive(:merge).and_return(pbi)
+
+          allow(patient).to receive(:slice)
+                              .with("registration_user_id", "registration_facility_id")
+                              .and_return(existing_patient.slice("registration_user_id",
+                                                                 "registration_facility_id"))
+          allow(patient).to receive_messages(id: existing_patient.id,
+                                             merge_status: :updated,
+                                             deleted_at: deleted_time,
+                                             update: patient)
+
+          allow(patient).to receive(:address=).and_return(existing_patient.address)
```

Comment:
Could we put this in the `receive_messages` above? `"address=": existing_patient.address`?